design/20210715.bundle-crd.md

```diff
@@ -0,0 +1,275 @@
+static, and can't be realistically changed in response to runtime events.
+
+cert-manager exposes a
+[`Status.CA` field on CertificateRequests](https://cert-manager.io/docs/concepts/certificaterequest/),
+as well as stores a
+[`ca.crt` field in generated Secrets resulting from Certificates](https://cert-manager.io/docs/concepts/certificate/).
+These values are however: best-effort by the Issuer and not required for a
+successful issuance; only available after a successful certificate request and
+scoped to that namespace; regarded as a controversial inclusion to the
+cert-manager API. This field only ever in practice contains one CA certificate,
+making managing multiple CA certificates or bundles difficult or impossible.
+Services often need multiple CA certificates as part of the same bundle or file.
+
+The upstream
+[Kubernetes CertificateSigningRequest resource](https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/)
+does not include any CA field. This means consumers after forced,
+[and are intended to](https://kubernetes.io/docs/reference/access-authn-authz/certificate-signing-requests/#kubernetes-signers),
+find an out of band method of propagating CA data.
+
+Some users have success by baking in CA data within built container images.
+This however requires the assumption that CA data is: known before runtime, long
+lived, and all images that require the CA data are built in house.
+
+Projects such as [Istio](https://istio.io/) (as well as Kubernetes itself) has
+had success with distributing trust roots, at runtime, by using a
+[well-known ConfigMap `istio-ca-root-cert`](https://istio.io/latest/docs/tasks/security/cert-management/plugin-ca-cert/)
+that is available within every namespace in a cluster. Bundle is an extension of
+this idea- to allow users to use the same approach with a resource that is
+expressive enough to meet most needs.
+
+## Goals
+
+- Create a `Bundle` resource to distribute trust bundles to be made available to
+    all services that need them.
+- Assist and help investigate
+  [upstream](https://github.com/kubernetes/kubernetes/issues/63726) the best
+  solution for trust distribution in Kubernetes.
+- Enable _most_ users to effectively distribute trust data in their clusters.
+
+
+### Non-Goals
+- Enable _every_ user's use case.
+- Compete with upstream development.
+
+
+## Proposal
+
+### Bundle
+
+A single resource `Bundle` within the `trust.cert-manager.io` group makes up
+the trust operator. This resource fetches and appends data from a number of
+`sources` in the trust namespace, and distributes them to a `target`.
+
+The trust namespace is the same namespace that the trust operator is deployed
+to, which would normally be the `cert-manager` namespace.
+
+While sources currently support `ConfigMap`, `Secret` and `InLine`, it could be
+extended to other resources in future, or even remote servers in the aid of
+custom informers.
```

Comment:
In two minds about this.

While I absolutely agree this would be a great addition, I feel we run the risk of;
1. The baked in trust bundles becoming out of date meaning the tool has an expiry data if left unmaintained,
2. Open a flood gate of "also add this default trust store" feature creep.